.github/workflows/release.yml

```diff
@@ -17,6 +17,33 @@ jobs:
       - uses: actions/checkout@v3
       - id: go-version
         run: echo "::set-output name=version::$(cat ./.go-version)"
+  changelog:
```

Comment:
One unfortunate side-effect of including the changelog handling within the existing `release.yml` workflow is that it is triggered _after_ the release tag (e.g. `v1.2.3`) has been pushed to the repository. Therefore if you checkout the repository at the tag, it will not include an updated `CHANGELOG.md` file.

To mitigate this, we have a few options:

- Remove the changelog logic in this workflow and have maintainers do the `changie` steps manually before tagging the release so the repository is already updated. Not a great option, but it is an option.
- Move the changelog building to its own workflow file (e.g. `.github/workflows/changelog.yml`) that is triggered via [workflow_dispatch](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) (you click it in the GitHub Actions web interface). This could essentially run the same logic here, just substituting `github.ref_name` with a "version" input. This would match the manual workflow we use in the terraform-plugin-* repositories today with go-changelog, just with an extra button to do it.
- Rework how the existing `release.yml` is triggered (it'd also be workflow_dispatch with a "version" input) so it does these steps in order:
    - Batches and merges the changelog, ensuring the repository is updated
    - Tags the version
    - Runs goreleaser for the rest of the build steps (release notes being `changes/VERSION.md`, which is _very_ nice)

I think we may prefer the reworked release workflow so no one has to think about extra steps when cutting a release, but curious your thoughts.

Reply:
I agree with you that the third option might be the best. I think triggering the release process should be simple and option two will add extra steps. Also changing the release process to be triggered via workflow_dispatch might be less error-prone than our current system of pushing git tags.